pkg/cache/file_based.go

```diff
@@ -1,58 +1,109 @@
 package cache
 
 import (
+	"bufio"
 	"fmt"
 	"os"
 	"path/filepath"
+	"strings"
 
 	"github.com/adrg/xdg"
-	"github.com/k8sgpt-ai/k8sgpt/pkg/util"
 )
 
 var _ (ICache) = (*FileBasedCache)(nil)
 
-type FileBasedCache struct{}
+type FileBasedCache struct {
+	noCache bool
+}
+
+func (f *FileBasedCache) IsCacheDisabled() bool {
+	return f.noCache
+}
 
-func (*FileBasedCache) Exists(key string) bool {
-	path, err := xdg.CacheFile(filepath.Join("k8sgpt", key))
+func (*FileBasedCache) Load(key string) (string, error) {
+	path, err := xdg.CacheFile(filepath.Join("k8sgpt", "analysis"))
 
 	if err != nil {
-		fmt.Fprintln(os.Stderr, "warning: error while testing if cache key exists:", err)
-		return false
+		return "", err
 	}
 
-	exists, err := util.FileExists(path)
-
+	file, err := os.Open(path)
 	if err != nil {
-		fmt.Fprintln(os.Stderr, "warning: error while testing if cache key exists:", err)
-		return false
+		if os.IsNotExist(err) {
+			// Créer le fichier s'il n'existe pas
+			file, err = os.Create(path)
+			if err != nil {
+				return "", err
+			}
+		} else {
+			return "", err
+		}
+	}
+	defer file.Close()
+
+	scanner := bufio.NewScanner(file)
+	for scanner.Scan() {
+		line := scanner.Text()
+		parts := strings.Split(line, ":")
+		// Faites quelque chose avec la ligne (ex: traitement des données)
+		if len(parts) == 2 && strings.TrimSpace(parts[0]) == strings.TrimSpace(key) {
+			return strings.TrimSpace(parts[1]), nil
+		}
 	}
 
-	return exists
+	if err := scanner.Err(); err != nil {
+		return "", err
+	}
+
+	return "", nil
 }
 
-func (*FileBasedCache) Load(key string) (string, error) {
-	path, err := xdg.CacheFile(filepath.Join("k8sgpt", key))
+func (*FileBasedCache) Store(key string, data string) error {
+	path, err := xdg.CacheFile(filepath.Join("k8sgpt", "analysis"))
+	if err != nil {
+		return err
+	}
 
+	file, err := os.OpenFile(path, os.O_RDWR|os.O_CREATE, 0600)
 	if err != nil {
-		return "", err
+		return err
+	}
+	defer file.Close()
+
+	var lines []string
+	exists := false
+	pattern := fmt.Sprintf("%s:%s", key, data)
+
+	scanner := bufio.NewScanner(file)
+	for scanner.Scan() {
+		line := scanner.Text()
+		if strings.HasPrefix(line, key+":") {
+			lines = append(lines, pattern)
+			exists = true
+		} else {
+			lines = append(lines, line)
+		}
 	}
 
-	data, err := os.ReadFile(path)
+	if err := scanner.Err(); err != nil {
+		return err
+	}
 
-	if err != nil {
-		return "", err
+	if !exists {
+		lines = append(lines, pattern)
 	}
 
-	return string(data), nil
-}
+	file.Truncate(0)
```

Comment:
I do not fully understand why the file is truncated and the written again. 
Could you elaborate a bit more?